hera_pspec/flags.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+mask_generator(nsamples, flags, n_threshold, greedy=False, axis, greedy_threshold, retain_flags=True):
+    """
+    Generates a greedy flags mask from input flags and nsamples arrays
+    
+    Parameters
+    ----------
+    nsamples : numpy.ndarray
+        integer array with number of samples available for each frequency channel at a given LST angle
+    
+    flags : numpy.ndarray
+        binary array with 1 representing flagged, 0 representing unflagged
+        
+    n_threshold : int
+        minimum number of samples needed for a point to remain unflagged
+        
+    greedy : bool
+        greedy flagging is used if true (default is False)
+        
+    axis : int
+        which axis to flag first if greedy=True (1 is row-first, 0 is col-first)
+        
+    greedy_threshold : float
+        if greedy=True, the threshold used to flag rows or columns if axis=1 or 0, respectively
+        
+    retain_flags : bool
+        LST-Bin Flags are left flagged even if thresholds are not met (default is True)
+    
+    Returns
+    -------
+    mask : numpy.ndarray
+	binary array of the new mask where 1 is flagged, 0 is unflagged
+
+"""
+
+    shape = nsamples.shape
+    flags_output = np.zeros(shape)
+
+    num_exactly_equal = 0
+
+    # comparing the number of samples to the threshold 
+    
+    for i in range(shape[0]):
```

Comment:
Can this nested `for` loop perhaps be replaced by a couple of strategic calls to `np.where`? We can discuss this.